public/app/core/components/NestedFolderPicker/NestedFolderList.tsx

```diff
@@ -15,31 +15,43 @@ import { FolderUID } from './types';
 
 const ROW_HEIGHT = 40;
 const CHEVRON_SIZE = 'md';
+const ID_PREFIX = 'folder-picker-item';
+
+export const getDOMId = (id: string) => `${ID_PREFIX}-${id ?? 'root'}`;
```

Comment:
the one 'problem' (that isn't actually a problem) is that these IDs are shared across all instances of NestedFolderPicker (they aren't globally unique), but you can't have multiple folder pickers open at the same time so in practice I think this isn't a real issue.